pkg/controller/nodes/transformers.go

```diff
@@ -82,13 +91,28 @@ func ToNodeExecutionEvent(nodeExecID *core.NodeExecutionIdentifier, info handler
 		OccurredAt: occurredTime,
 	}
 
-	// TODO this should use node-node relationship instead of taskID
-	if status.GetParentTaskID() != nil {
+	if eventVersion == v1alpha1.V0 && status.GetParentTaskID() != nil {
 		nev.ParentTaskMetadata = &event.ParentTaskExecutionMetadata{
 			Id: status.GetParentTaskID(),
 		}
 	}
 
+	if eventVersion != v1alpha1.V0 {
```

Comment:
I know it does not exist, but can you add a few unit tests for this code?

Reply:
No unit tests have been added. I am on it.